x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java

```diff
@@ -136,14 +137,16 @@ public void testDoLogWithDebug() throws IOException {
             Map<String, String> colA = Map.of("name", "DO_LOG_ME", "type", "integer");
             assertEquals(List.of(colA), result.get("columns"));
             assertEquals(List.of(List.of(1)), result.get("values"));
-            try (InputStream log = cluster.getNodeLog(0, LogType.SERVER)) {
-                boolean[] found = new boolean[] { false };
-                Streams.readAllLines(log, line -> {
-                    if (line.contains("DO_LOG_ME")) {
-                        found[0] = true;
-                    }
-                });
-                assertThat(found[0], equalTo(true));
+            boolean[] found = new boolean[] { false };
+            for (int i = 0; i < cluster.getNumNodes(); i++) {
+                try (InputStream log = cluster.getNodeLog(0, LogType.SERVER)) {
+                    Streams.readAllLines(log, line -> {
+                        if (line.contains("DO_LOG_ME")) {
+                            found[0] = true;
+                        }
+                    });
+                    assertThat(found[0], equalTo(true));
+                }
             }
```

Comment:
Do we expect logs on all the nodes? Shouldn't the assertion be after the for loop?
```suggestion
                }
            }
            assertThat(found[0], equalTo(true));
```

Reply:
> Do we expect logs on all the nodes? Shouldn't the assertion be after the for loop? 

The `test-update-serverless` label should have found this.

Reply:
For science!